lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.module

```diff
@@ -38,6 +38,41 @@ function _dosomething_gladiator_build_http_client() {
 }
 
 /**
+ * Check if the given user is already in a competition for this campaign.
+ * If they are in a competition or waiting room in Gladiator, update their
+ * local signup record to reflect that.
+ *
+ * @param $signup - signup entity
+ *
+ * @return bool
+ */
+function dosomething_gladiator_in_a_competition($signup) {
+  $client = _dosomething_gladiator_build_http_client();
+
+  $query = drupal_http_build_query([
+    'id' => dosomething_user_get_northstar_id($signup->uid),
+    'campaign_id' => $signup->nid,
+    'campaign_run_id' => $signup->run_nid,
+  ]);
+
+  $response = drupal_http_request($client['base_url'] . '/users?'.$query, [
+    'method' => 'GET',
+    'headers' => $client['headers'],
+  ]);
+
+  $json = json_decode($response->data, TRUE);
+  $already_in_gladiator = !empty($json['data']['waitingRoom']) || !empty($json['data']['competition']);
```

Comment:
Nope, `empty` won't give a warning if the variable or key doesn't exist. Same deal as with `isset`.